sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/FromUnsafe.scala

```diff
@@ -52,6 +53,8 @@ case class FromUnsafe(child: Expression) extends UnaryExpression
       }
       new GenericArrayData(result)
 
+    case StringType => value.asInstanceOf[UTF8String].clone()
```

Comment:
Yes, I know, I assume we need to handle the top data type in `StringType` also, otherwise, it still will cause confusing if people call the `FromUnsafe` to get the `StringType` of data.

I mean we'd better add the `StringType` support in https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Projection.scala#L159